File: app/src/main/java/de/tum/in/tumcampusapp/models/tumcabe/ChatRoom.java
package de.tum.in.tumcampusapp.models.tumcabe;

public class ChatRoom {

    private String name;
    private int id;

    private int members = -1;

    public ChatRoom(String name) {
        super();
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getMembers() {
        return members;
    }

    public void setMembers(int members) {
        this.members = members;
    }

    @Override
    public String toString() {
        return id + ": " + name;
    }
}

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/chat/model/ChatRoom.kt
package de.tum.`in`.tumcampusapp.component.ui.chat.model

import de.tum.`in`.tumcampusapp.R.string.name

data class ChatRoom(var title: String = "",
                    var semester: String = "ZZZ") {
    var id: Int = 0
    var members = -1

    constructor(combined: String) : this() {
        if (combined.contains(":")) {
            semester = combined.substring(0, 3)
            title = combined.substring(4)
        }
    }

    fun getCombinedName(): String {
        return "$semester:$title"
    }

    override fun toString() = "$id: $name"

    companion object {
        @JvmField
        val MODE_JOINED = 1
        @JvmField
        val MODE_UNJOINED = 0
    }

}
